tests/infra/remote.py

```diff
@@ -213,6 +220,10 @@ def get(self, file_name, dst_path, timeout=FILE_TIMEOUT, target_name=None):
                         if os.path.exists(dst_dir):
                             shutil.rmtree(dst_dir)
                         os.makedirs(dst_dir)
+                        if pre_condition_func is not None:
+                            pre_condition_func(
```

Comment:
I would expect a pre_condition_func to cause an operation to abort if it returns false, I guess the idea is to raise here instead?